### ISSUE TITLE
The plugin causes strapi import to fail 

### ISSUE THREAD

**Issue by Ahmedsaed** [pluginpal/strapi-plugin-sitemap#174]
### Describe the bug

Whenever I try to import a backup made using strapi export. I get this error

```
error: [strapi-plugin-sitemap]: Something went wrong while trying to build the SitemapStream. TypeError: Invalid URL
```

### Steps to reproduce the behavior

1. install and configure the plugin
2. run `npm run strapi export` at source machine
3. run `npm run strapi import` at destination machine 

### Expected behavior

I expect the import to complete sucessfully

### System

- Node.js version: v20.12.2
- NPM version: 10.5.0
- Strapi version: v4.25.1 and v4.23.1
- Plugin version:  3.1.1 and 3.8.0
- Database: postgres
- Operating system: linux

**Comment by Ahmedsaed** [comment]
This issues also affects `strapi transfer` with the same error

**Comment by HemantDubey-ACS** [comment]
We are also getting the same error while `strapi transfer`.

**Comment by Ahmedsaed** [comment]
I have a workaround. Just delete all sitemap settings defined in admin panel in the **DESTINATION MACHINE**. Effectively resetting the plugin to a state where it doesn't initialize the `sitemapStream`.

Regarding the issue itself, The error message is written in `createSitemap` function in `/server/services/core.js`. I am not sure what triggers this function during a strapi transfer or import so I can't recommend a fix.

Either prevent the function from being triggered or add some conditions to handle this case.

**Comment by MSACC** [comment]
> I have found a workaround. Just delete all sitemap settings defined in admin panel in the **DESTINATION MACHINE**. Effectively resetting the plugin to a state where it doesn't initialize the `sitemapStream`.
> 
> Regarding the issue itself, The error message is written in `createSitemap` function in `/server/services/core.js`. I am not sure what triggers this function during a strapi transfer or import so I can't recommend a fix.
> 
> Either prevent the function from being triggered or add some conditions to handle this case.

Thanks for investigating this @Ahmedsaed . If you could help us find the solution, or maybe send in a PR when found, we can add it as a fix for you. For now we are not planning to develop a fix / investigate further soon, as you have found a workaround for it. When having a fix, your help will be very appreciated.

**Comment by boazpoolman** [comment]
I'm still uncertain why the create sitemap service is called with Strapi Transfer.
It could be that you (@Ahmedsaed and @HemantDubey-ACS) have the `autoGenerate` setting enabled?

**Comment by Ahmedsaed** [comment]
> I'm still uncertain why the create sitemap service is called with Strapi Transfer.
> Could it be that you (@Ahmedsaed and @HemantDubey-ACS) have the `autoGenerate` setting enabled?

Yea, `autoGenerate` was enabled.

Here is my config:
```
sitemap: {
    enabled: true,
    config: {
      limit: 45000,
      xsl: true,
      autoGenerate: true,
      caching: true,
      allowedFields: ["id", "uid", "slug"],
      excludedTypes: [],
    },
  },
```

**Comment by boazpoolman** [comment]
That makes sense then.

Whenever an import occurs with Strapi Transfer, entries are created, and thus the auto generate features kicks in to build the sitemap. Regardless if there has been a hostname set or not.

I've attempted to solve the issue in #178
Just adding some checks on the hostname like you suggested Ahmed.

**Comment by boazpoolman** [comment]
This issue has been fixed and released with version [3.2.0](https://github.com/pluginpal/strapi-plugin-sitemap/releases/tag/3.2.0)
